tests/scm.py

```diff
@@ -59,3 +59,13 @@ def test_prep_tree_for_archive(self):
         scm_base.prep_tree_for_archive("test1", basedir, "test2")
 
         self.assertEqual(scm_base.arch_dir, os.path.join(basedir, "test2"))
+
+    def test_version_iso_cleanup(self):
+        # Avoid get_repocache_hash failure in Scm.__init__
+        self.cli.url = ""
+        scm_base = Scm(self.cli, None)
+        self.assertEqual(scm_base.version_iso_cleanup("2.0.1-3"), "2.0.1-3")
+        self.assertEqual(scm_base.version_iso_cleanup("1"), "1")
+        self.assertEqual(scm_base.version_iso_cleanup("1.54-1.2"), "1.54-1.2")
+        self.assertEqual(scm_base.version_iso_cleanup("2017-01-02 02:23 +10"),
+                         "20170102 0223 +10")
```

Comment:
@aspiers these are just some basic tests, including the one that would fail without the fix (first one), let me know if you want to add some more